content/en/blog/_posts/2023-12-20-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: blog
+title: "Removing In-Tree Cloud Providers Graduates to Beta Status"
+date: 2023-12-20
+slug: removing-in-tree-cloud-providers-beta-status
+---
+
+**Authors:** Michael McCune, Andrew Sy Kim
+
+In 2018, the Kubernetes community agreed to form the Cloud Provider Special Interest Group (SIG), with a mission to externalize all cloud provider integrations and remove all the existing in-tree cloud provider integrations. In January 2019, the Kubernetes community approved the initial draft of [KEP-2395: Removing In-Tree Cloud Provider Code][kep2395]. This KEP defines a process by which we can remove cloud provider specific code from the core Kubernetes source tree. From the KEP:
+
+> Motiviation behind this effort is to allow cloud providers to develop and make releases independent from the core Kubernetes release cycle. The de-coupling of cloud provider code allows for separation of concern between "Kubernetes core" and the cloud providers within the ecosystem. In addition, this ensures all cloud providers in the ecosystem are integrating with Kubernetes in a consistent and extendable way.
+
+After many years of development and collaboration across many contributors, SIG Cloud Provider is happy to announce that as of the Kubernetes 1.29 release, [KEP-2395][kep2395] will graduate to Beta status. The graduation to Beta is a big milestone as we’ve had to work through the complexities that come with migrating a core component in a distributed system like Kubernetes. With the graduation to Beta, all Kubernetes core components will have cloud provider integrations disabled by default. This brings us one step closer to its final removal which will happen when the KEP graduates to GA.
+
+Although the new defaults will affect all Kubernetes users, they will require changes for users who are currently using the in-tree providers for Azure, GCE, and vSphere.
```

Comment:
I'd expect to see something like:
```
These updated default settings affect a large proportion of Kubernetes users, and
**will require changes** for users who were previously using the in-tree provider integrations.
The legacy integrations offered compatibility with for Azure, AWS, GCE, OpenStack and vSphere; however,
for AWS and OpenStack DETAILS_GO_HERE.
```

I know there is a legacy in-tree AWS integration, so it's odd that we don't mention it here. Same for OpenStack.

(ie: don't make readers skip past this part of the page to understand what's going on)

Reply:
i specifically didn't mention the AWS or OpenStack providers here because they were removed prior to 1.27. but i will add them in to make it more clear.

Reply:
updated